src/core/ext/filters/client_channel/lb_policy/xds/xds_load_balancer_api.cc

```diff
@@ -18,12 +18,129 @@
 
 #include <grpc/support/port_platform.h>
 
-#include "pb_decode.h"
-#include "pb_encode.h"
 #include "src/core/ext/filters/client_channel/lb_policy/xds/xds_load_balancer_api.h"
 
 #include <grpc/support/alloc.h>
 
+namespace grpc_core {
+
+namespace {
+
+constexpr char kEdsTypeUrl[] =
+    "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment";
+constexpr char kEndpointRequired[] = "endpointRequired";
+upb_alloc* g_arena = nullptr;
+
+void InitArena() { g_arena = upb_arena_new(); }
+
+upb_alloc* arena() {
```

Comment:
Are you sure you want a global arena? None of the protos will ever be deleted. This doesn't seem like the right choice.

Reply:
I need to call `upb_arena_free()` at some point to free the protos, right? How frequently should I use a separate arena? Per RPC?

Reply:
I think per RPC is probably the most natural lifetime for an arena. Do you have other resources associated with the RPC that you free at the end?